dandi/upload.py

```diff
@@ -335,3 +341,22 @@ def upload_agg(*ignored):
                 else:
                     rec.update(skip_file(exc))
             out(rec)
+
+    if sync:
+        relpaths = []
+        for p in original_paths:
+            rp = os.path.relpath(p, dandiset.path)
+            relpaths.append("" if rp == "." else rp)
+        path_prefix = reduce(os.path.commonprefix, relpaths)
+        to_delete = []
+        for asset in client.get_dandiset_assets(
+            ds_identifier, "draft", path=path_prefix
+        ):
+            if (
+                any(p == "" or path_is_subpath(asset["path"], p) for p in relpaths)
+                and not Path(dandiset.path, asset["path"]).exists()
+            ):
+                to_delete.append(asset["asset_id"])
+        if to_delete and click.confirm(f"Delete {len(to_delete)} assets on server?"):
```

Comment:
do the same pluralism here as I suggested for download.  could even be made into a helper function in tools to avoid duplication

Reply:
Done.